### ISSUE TITLE
Simplify `purs publish` resolutions format

### ISSUE THREAD

**Issue by hdgarrood** [purescript/purescript#3499]
Currently, `purs publish` expects to be supplied with two JSON files: a manifest file and a resolutions file. The latter is only used to determine the precise versions of all dependencies which are in use on the developer's machine, so that we can generate links across packages which work (without having to do any version solving on the Pursuit end). The compiler expects the resolutions file to have the same schema as the output of `bower list --offline --json`, but this schema is hugely wasteful -- so wasteful, in fact, that we're seeing OOM errors in the course of trying to produce the JSON file: see https://github.com/purescript-contrib/pulp/issues/351. A much simpler schema would suffice, e.g.:

```
// resolutions.json
{
  "purescript-prelude": "4.0.0",
  "purescript-lists": "6.0.0",
  ...
}
```

and we would leave the responsibility of producing this data to external tools (as is the case currently). There are two additional upsides of doing this:

1. It further decouples the compiler from Bower; technically, any tool *could* produce this data according to the schema we're expecting, but realistically, because it's such a silly schema, none actually will (apart from Bower). It will be much easier for alternative tools to provide this data if they're able to use a more sensible schema.
2. It allows us to simplify the part of the compiler which currently extracts the relevant information from the resolutions file.

**Comment by hdgarrood** [comment]
I've realised that implementing this will mean that `purs publish` will lose the ability to warn you about undeclared or extraneous dependencies, as we currently use the structure of the resolutions json file to determine whether there are any such dependencies. However I am not too worried about this: I don't think it should be the responsibility of `purs publish` anyway. Arguably we should have removed this feature way back when we decided to decouple the compiler from Bower.
